#! /usr/bin/env python

PACKAGE='twist_controller'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                 Type         Description                      Default   Min     Max
gen.add("decel_max",        double_t, 0,  "Deceleration limit(m/s^2)",       5.0,    0.0,    5.0)
gen.add("accel_max",        double_t, 0,  "Acceleration limit(m/s^2)",       1.0,    0.0,    5.0)
gen.add("steer_kp",         double_t, 0,  "Steering feedback using gyro",    3.0,    0.0,    50.0)

gen.add("speed_kp",         double_t, 0, "Speed loop proportional gain",     2.0,    0.0,    50.0)
gen.add("speed_ki",         double_t, 0, "Speed loop integral gain",         1.0,    0.0,    50.0)
gen.add("speed_kd",         double_t, 0, "Speed loop derivation gain",       0.0,    0.0,    50.0)

gen.add("accel_kp",         double_t, 0, "Accel loop proportional gain",     0.4,    0.0,    8.0)
gen.add("accel_ki",         double_t, 0, "Accel loop integral gain",         0.2,    0.0,    8.0)
gen.add("accel_kd",         double_t, 0, "Accel loop derivation gain",       0.2,    0.0,    8.0)

exit(gen.generate(PACKAGE, PACKAGE, "Controller"))
